Automattic/wp-calypso
# Editor: use redux post-counts in drafts button.

With the addition of `post-counts` to the redux state tree, this branch utilizes that data source to power the drafts button in the Editor drawer.

<img width="341" alt="new_post_ _trout_bummin_ _wordpress_com" src="https://cloud.githubusercontent.com/assets/22080/14267738/10b8c9bc-fa88-11e5-9961-17108f3e1da8.png">

__To Test__
- Open up a post in the editor for a wpcom site
- Ensure the proper drafts number shows up in the drawer button and you can open the drawer if there any drafts exist ( draft count >= 1 )
- Open up a post in the editor for a jetpack site
- Verify to draft count is displayed and you can open the draft drawer

__Notes__
It appears this was the last place using `components/data/post-counts-data`.  I considered deleting that component as part of this PR but curious to hear thoughts on that @aduth
